admin/server/auth/handlers.go

```diff
@@ -674,15 +680,17 @@ func (a *Authenticator) getAccessToken(w http.ResponseWriter, r *http.Request) {
 	}
 
 	grantType := values.Get("grant_type")
-	if !(grantType == deviceCodeGrantType || grantType == authorizationCodeGrantType) {
+	if !(grantType == deviceCodeGrantType || grantType == authorizationCodeGrantType || grantType == refreshTokenGrantType) {
 		http.Error(w, "invalid grant_type", http.StatusBadRequest)
 		return
 	}
 
 	if grantType == deviceCodeGrantType {
 		a.getAccessTokenForDeviceCode(w, r, values)
-	} else {
+	} else if grantType == authorizationCodeGrantType {
 		a.getAccessTokenForAuthorizationCode(w, r, values)
+	} else {
+		a.getAccessTokenForRefreshToken(w, r, values)
```

Comment:
To be defensive, maybe add an explicit `else if grantType == refreshTokenGrantType`, and then have a last `else` case that errors with `unexpected grant type %q`?